service/src/main/java/bio/terra/workspace/service/iam/SamService.java

```diff
@@ -195,12 +195,16 @@ public void createWorkspaceWithDefaults(AuthenticatedUserRequest userRequest, UU
     ResourcesApi resourceApi = samResourcesApi(userRequest.getRequiredToken());
     // Sam will throw an error if no owner is specified, so the caller's email is required. It can
     // be looked up using the auth token if that's all the caller provides.
-    String callerEmail =
-        Optional.ofNullable(userRequest.getEmail()).orElse(getUserEmailFromSam(userRequest));
+
+    // If we called WSM as the pet SA and went through the proxy, this becomes the pet SA's email if
+    // we use the pet email. That caused an issue where the human user wasn't recognized on the
+    // workspace.
+
+    String humanUserEmail = getUserEmailFromSam(userRequest);
     CreateResourceRequestV2 workspaceRequest =
         new CreateResourceRequestV2()
             .resourceId(id.toString())
-            .policies(defaultWorkspacePolicies(callerEmail));
+            .policies(defaultWorkspacePolicies(humanUserEmail));
```

Comment:
Are there other places WSM sets an email as a SAM resource owner? I was thinking of maybe controlled resources. But not sure if this applies to child resources also.

Reply:
For controlled resources, the parent is the fully qualified workspace ID. In `bio.terra.workspace.service.iam.SamService#createControlledResource`, the Api is created from the user associated with the token in the user request.  For private resources, I've changed it to use the user email from Sam instead of the request email.